includes/wikia/parser/templatetypes/handlers/NavigationTemplate.class.php

```diff
@@ -85,6 +85,18 @@ private static function replaceMarker( $marker, $html ) {
 	private static function mark( $text ) {
 		// marking each template with unique marker to be able to handle nested navigation templates
 		$marker = "\x7f" . self::MARK . "_" . uniqid() . "\x7f";
-		return sprintf( "<%s>%s</%s>", $marker, $text, $marker );
+		return sprintf( "<%s>\n%s\n</%s>", $marker, $text, $marker );
+	}
+
+	public static function removeInnerMarks( $templateWikitext ) {
+		$openMarkerRegex = "[<&lt;]\x7f" . self::MARK . "[^\x7f]+\x7f[>&gt;]\\n";
+		$closeMarkerRegex = "\\n[<&lt;]\\/\x7f" . self::MARK . ".+\x7f[>&gt;]";
+
+		if (preg_match( "/(" . $openMarkerRegex . ")(.*)(" . $closeMarkerRegex . ")/is", $templateWikitext, $inside )) {
+			$replacedOpenings = preg_replace( "/" . $openMarkerRegex . "/isU", "", $inside[2] );
```

Comment:
Should new lines be removed here as well? What if regex fails? Is it possible to make those replaces in one?